Signal/src/ViewControllers/ContactsViewHelper.m

```diff
@@ -199,6 +207,76 @@ - (BOOL)doesSignalAccount:(SignalAccount *)signalAccount matchSearchTerms:(NSArr
         }]];
 }
 
+- (BOOL)doesContact:(Contact *)contact matchSearchTerm:(NSString *)searchTerm
+{
+    OWSAssert(contact);
+    OWSAssert(searchTerm.length > 0);
+
+    if ([contact.fullName.lowercaseString containsString:searchTerm.lowercaseString]) {
+        return YES;
+    }
+
+    NSString *asPhoneNumber = [PhoneNumber removeFormattingCharacters:searchTerm];
+    if (asPhoneNumber.length > 0) {
+        for (PhoneNumber *phoneNumber in contact.parsedPhoneNumbers) {
+            if ([phoneNumber.toE164 containsString:asPhoneNumber]) {
+                return YES;
+            }
+        }
+    }
+
+    return NO;
+}
+
+- (BOOL)doesContact:(Contact *)contact matchSearchTerms:(NSArray<NSString *> *)searchTerms
+{
+    OWSAssert(contact);
+    OWSAssert(searchTerms.count > 0);
+
+    for (NSString *searchTerm in searchTerms) {
```

Comment:
We do something similar when searching for `SignalContacts`.  There's just enough differences that it doesn't make sense to de-duplicate, IMO.